packages/nimble-components/src/list-option/index.ts

```diff
@@ -51,6 +52,17 @@ export class ListOption extends FoundationListboxOption {
             .map(node => node.textContent?.trim())
             .join(' ');
     }
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        if (this.isDropdownOwner(this.parentElement)) {
```

Comment:
When we do #791 do you think we'll end up with `list-option`s that aren't direct children of a `select`/`combobox`? Anything to think through or write tests for now? Or do we just need to try to remember to think about this when we work on that?

For example:
```html
<nimble-select>
   <nimble-list-group>Texas
      <nimble-list-option>Austin</nimble-list-option>
      <nimble-list-option>San Antonio</nimble-list-option>
   </nimble-list-group>
   <nimble-list-group>Not Texas
      <nimble-list-option>Tokyo</nimble-list-option>
   </nimble-list-group>
</nimble-select>
```

But then again we might [copy Shoelace's flat pattern](https://shoelace.style/components/select#grouping-options). Or not, it might make it hard to filter by all options in a section.

```html
<sl-select>
  <small>Section 1</small>
  <sl-option value="option-1">Option 1</sl-option>
  <sl-option value="option-2">Option 2</sl-option>
  <sl-option value="option-3">Option 3</sl-option>
  <sl-divider></sl-divider>
  <small>Section 2</small>
  <sl-option value="option-4">Option 4</sl-option>
  <sl-option value="option-5">Option 5</sl-option>
  <sl-option value="option-6">Option 6</sl-option>
</sl-select>
```

Reply:
> But then again we might copy Shoelace's flat pattern

I think our goal should be to copy patterns from DOM first, i.e. the former with a group element: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup

Reply:
I prefer we not bother with making this implementation robust enough to handle the inevitable feature that #791 will introduce, as we still don't have an agreed upon design in place for handling that.